### PR TITLE
Check submission table when computing submission history

### LINKED ISSUE #15822: Submission History reads azure storage table when report not found in postgresql database
### User Story

As a sender to the new submissions api
I want to use the submissions history api, 
so that I can view the history for all successful and failed submissions.

### Description/Use Case
The submissions api does not connect to the postgresql database. Instead, it inserts to a table in the azure storage account. That table has information includes submission status and details which could include error information. The information in that table should be used in the submission history response.

### Dev Notes
The submissions api currently inserts to the submission table in the azure storage account(_not in postgresql_) and uploads a file to the blob storage. Then a queue message is added to the FHIRReceive queue. The current submission history expects records to exist for action and report_file. Since the submission micro service does not connect to postgresql it cannot verify the client_id supplied in the submission. The FHIRReceive step does the verification. If a sender is not found, then no record is made of the submission in postgresql. The only records are in the storage account table.

With that in mind, the submission history needs to be updated to review that table when no report exists in the postgresql database.

### Acceptance Criteria
- [ ] When a report is not in postgresql the submission history responds appropriately with detail from submission table

### PR DESCRIPTION
**Pull request by thetaurean**
This PR updates the submission history code to check the new submission table in azure storage when a submissionId is not found in the primary postgres database.

Test Steps:
1. Run integration test suite

## Changes
- Updated submission history code to check azure storage table
- Added timestamp field and JSON annotations to shared `Submission` class

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [x] Added tests?

## Linked Issues
- Fixes #15822

### CODE REVIEW

**Review by david-navapbc — CHANGES_REQUESTED** [review]
there is a test failure when I run this on my workstation that does not happen when I do the same in master branch.

```java.lang.NullPointerException: getenv(...) must not be null
      at gov.cdc.prime.router.history.azure.DeliveryFunctionTests.test delivery details(DeliveryFunctionTests.kt:465)
```

**Review by david-navapbc — APPROVED** [review]
tests now pass for me locally - nice work!

File: prime-router/src/main/kotlin/history/azure/ReportFileFunction.kt [hunk 1]
```diff
@@ -161,10 +162,15 @@ abstract class ReportFileFunction(
...
-        } catch (ex: IllegalStateException) {
-            logger.error(ex)
-            // Errors above are actionId or UUID not found errors.
-            return HttpUtilities.notFoundResponse(request, ex.message)
+        } catch (ex: IllegalStateException) { // actionId or UUID not found
+            val submission = SubmissionTableService.getInstance().getSubmission(id, "Received")
+
+            return if (submission == null) {
+                logger.error(ex)
+                HttpUtilities.notFoundResponse(request, ex.message)
```
[the remark this answers is not in the document]
Comment (thetaurean) [reply prime-router/src/main/kotlin/history/azure/ReportFileFunction.kt:170]: I agree with you however this was pre-existing code so I didn't feel comfortable changing it.

For the most part this is catching exceptions that we throw, so the error messages are things we set. Though `IllegalStateException` is pretty broad and could probably occur unexpectedly.

**Review by arnejduranovic — APPROVED** [review]
Changes look good. In the background will work to put up a follow-up PR to use dependency injection with SubmissionTableService

**Pull request merged**